### PR TITLE
refactor(CORE/Instance): Move Pandemonius script from SAI to Core script

### PR DESCRIPTION
**Pull request by Ercules76**
Pandemonius (id: 18341) better performace removing SAI

Ported from TrinityCore

## CHANGES PROPOSED:
-  Removing Smart Script
-  Adding Core Script

## ISSUES ADDRESSED:
Isn't a real issue, it's a refactoring for better performance
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3105

## TESTS PERFORMED:
- Compiled at 64bit without errors and warnings
- Compiled on windows 10 with Visual Studio 16

## HOW TO TEST THE CHANGES:

Enter the instance and enjoy :)

## KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
Is there really a performance gain? Usually we should port things from core to sai, not the opposite :)

**Comment by Ercules76** [comment]
> 
> 
> Is there really a performance gain? Usually we should port things from core to sai, not the opposite :)

I think no, the only difference i've noticed is the scheduled event of a spell. I took for good what was said in the issue

**Comment by pklloveyou** [comment]
Tested, no problems

File: src/server/scripts/Outland/Auchindoun/ManaTombs/boss_pandemonius.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    CreatureAI* GetAI(Creature* creature) const
+    {
+        return new boss_pandemoniusAI(creature);
+    }
+
+    struct boss_pandemoniusAI : public ScriptedAI
+    {
+        boss_pandemoniusAI(Creature* creature) : ScriptedAI(creature)
+        {
+        }
```
Comment (Kitzunu) [inline src/server/scripts/Outland/Auchindoun/ManaTombs/boss_pandemonius.cpp:46]: What is this good for?

File: data/sql/updates/pending_db_world/rev_1593176049024213800.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1593176049024213800');
+
+DELETE FROM `smart_scripts` WHERE `entryorguid` IN (18341, 1834100, 1834101);
+UPDATE `creature_template` SET `AIName` = '', `ScriptName` = 'boss_pandemonius' WHERE `entry` = '18341';
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1593176049024213800.sql:4]:
```suggestion
UPDATE `creature_template` SET `AIName` = '', `ScriptName` = 'boss_pandemonius' WHERE `entry` = 1834;
```

File: src/server/scripts/Outland/Auchindoun/ManaTombs/boss_pandemonius.cpp (hunk 1 above)
Comment (Ercules76) [reply src/server/scripts/Outland/Auchindoun/ManaTombs/boss_pandemonius.cpp:46]: is a problem?

Comment (Kitzunu) [reply src/server/scripts/Outland/Auchindoun/ManaTombs/boss_pandemonius.cpp:46]: No I just wonder what the use of empty brackets are ^^

Comment (Ercules76) [reply src/server/scripts/Outland/Auchindoun/ManaTombs/boss_pandemonius.cpp:46]: are the brackets of the constructor

Comment (Shard-MW) [reply src/server/scripts/Outland/Auchindoun/ManaTombs/boss_pandemonius.cpp:46]: Maybe just put them inline like : 
https://github.com/azerothcore/azerothcore-wotlk/pull/3165/files/083b3c942ff8f26024d68b6324ef48c77b9f2dbe#diff-83ffc1c83be6eb3b673c01afcfc12e20R35

I find this better for (my) eyes hahaha

Comment (Ercules76) [reply src/server/scripts/Outland/Auchindoun/ManaTombs/boss_pandemonius.cpp:46]: ok

**Comment by illunix** [comment]
> (personal opinion)
> Bosses = core
> Monsters, rares, elites, rare elites = SAI
> 
> Hope to see this merged. 🐱

When should I go with LUA? I heard I never should go with that? @Si1ker

**Comment by Kitzunu** [comment]
You can only script in LUA if you use the Eluna module, which the "main" core does not support.
So you can code in LUA whenever you want for "custom" stuff / scripts for Eluna :)

**Comment by illunix** [comment]
I know it's not default, I'm just curious for what should I use LUA? I heard that if you have a lot of scripts written in lua it slows down the server. And I don't know what I should pick for example for scripting quest, lua, cpp or sai? etc.

**Comment by Kitzunu** [comment]
Ah, well I am no lua expert.
Better to ask in Discord in the #lua-scripts channel

**Pull request merged**